drivers/pppos.c

```diff
@@ -25,7 +25,52 @@
 #include <sys/msg.h>
 #include <syslog.h>
 
+/* Intentionally does not begin from 1. */
+#define PPPOS_MODEM_GENERIC 2
+#define PPPOS_MODEM_QUECTEL 3
+
+#if PPPOS_MODEM_TYPE == PPPOS_MODEM_GENERIC
+
+#define PPPOS_USE_CONFIG_FILE 1
+#define PPPOS_DISCONNECT_ON_INIT 1
+#define AT_CONNECT_CMD "AT+CGDATA=\"PPP\",1\r\n"
+#define AT_INIT_CMDS_TIMEOUT_MS 3000
+#define AT_CONNECT_CMD_TIMEOUT_MS 3000
+
+static const char* at_init_cmds[] = {
+	"ATZ\r\n",			// reset MODEM
+	"ATQ0 V1 E0 S0=0 &C1 &D2 +FCLASS=0\r\n",	// setup serial/message exchange
+	"AT+WS46=29\r\n",		// disable LTE
+	"AT+CREG?\r\n", 		// check network registration (just for debug)
+	"AT+COPS?\r\n", 		// check operator registration (just for debug)
+	"AT+CSQ\r\n",       // check signal quality (for debug)
+	NULL,
+};
+
+#elif PPPOS_MODEM_TYPE == PPPOS_MODEM_QUECTEL
+
+#define PPPOS_USE_CONFIG_FILE 0
+#define PPPOS_DISCONNECT_ON_INIT 0
+#define AT_CONNECT_CMD "AT+CGDATA=\"PPP\",1\r\n"
+#define AT_INIT_CMDS_TIMEOUT_MS 3000
+#define AT_CONNECT_CMD_TIMEOUT_MS 3000
+
+static const char* at_init_cmds[] = {
+	"ATZ\r\n",
```

Comment:
it would be great to add some comments what these AT commands are doing without looking at the documentation.

Note: missing ECHO off command we were discussing previously (this would clean up the logs a little)

Reply:
I have included the comments and also the ATE0 command (at least for Quectel).

Reply:
Thanks, for telit it's done in one multi-command specifying serial command specifics (`E0` part is "echo off").